src/modules/sync/utils/contentTypeHelpers.ts

```diff
@@ -0,0 +1,77 @@
+import { ContentTypeContracts, ContentTypeElements, ContentTypeSnippetContracts } from "@kontent-ai/management-sdk";
+import { PortableTextInternalLink, PortableTextObject, traversePortableText } from "@kontent-ai/rich-text-resolver";
+
+export const getGuidelinesElements = (
+  types: ContentTypeContracts.IContentTypeContract[] | ContentTypeSnippetContracts.IContentTypeSnippetContract[],
+) =>
+  types.flatMap(type =>
+    type.elements.filter(element => element.type === "guidelines")
+  ) as unknown as ContentTypeElements.IGuidelinesElement[];
+
+export const getAssetElements = (
+  types: ContentTypeContracts.IContentTypeContract[] | ContentTypeSnippetContracts.IContentTypeSnippetContract[],
+) =>
+  types.flatMap(type =>
+    type.elements.filter(element => element.type === "asset")
```

Comment:
okay me likey the idea :D

Reply:
well there is a problem with typegueard of guidelines because ElementContracts.IContentTypeElementContract contains name and ContentTypeElements.IGuidelinesElement does not :/